rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -152,10 +153,27 @@ std::shared_ptr<rosbag2_storage::SerializedBagMessage> SequentialReader::read_ne
   throw std::runtime_error("Bag is not open. Call open() before reading.");
 }
 
-std::vector<rosbag2_storage::TopicMetadata> SequentialReader::get_all_topics_and_types()
+const rosbag2_storage::BagMetadata & SequentialReader::get_metadata() const
 {
   if (storage_) {
-    return storage_->get_all_topics_and_types();
+    return metadata_;
+  }
+  throw std::runtime_error("Bag is not open. Call open() before reading.");
+}
+
+
+void SequentialReader::fill_topics_and_types()
+{
+  topics_metadata_.reserve(metadata_.topics_with_message_count.size());
+  for (const auto & topic_information : metadata_.topics_with_message_count) {
+    topics_metadata_.push_back(topic_information.topic_metadata);
+  }
+}
```

Comment:
I think this could be a free function (inline) which takes the metadata as input and returns the topics_metadata